Jenkinsfile

```diff
@@ -1,51 +1,177 @@
-// this will need to be pipeline-library@master when the pr merges
-@Library('pipeline-library@ui-pipeline-legacy') _
+import groovy.json.JsonBuilder
+
+@Library('pipeline-library@COMUI-857') _
+
+def MAIN_BRANCH = 'master'
+def DEVELOP_BRANCH = 'develop'
+
+def isMain = {
+  env.BRANCH_NAME == MAIN_BRANCH
+}
+
+def isRelease = {
+  env.BRANCH_NAME.startsWith('release/')
+}
+
+def isDevelop = {
+  env.BRANCH_NAME == DEVELOP_BRANCH
+}
+
+def getBuildType = {
+  isMain()
+    ? 'MAINLINE'
+    : 'FEATURE'
+}
 
 webappPipeline {
-    slaveLabel = 'dev_v2'
-    useArtifactoryRepo = false
     projectName = 'developercenter-cdn/webrtc-sdk'
+    team = 'Genesys Client Media (WebRTC)'
+    mailer = 'purecloud-client-media@genesys.com'
+    chatGroupId = '763fcc91-e530-4ed7-b318-03f525a077f6'
+
+    nodeVersion = '14.x'
+    buildType = getBuildType
+
     manifest = customManifest('dist') {
         sh('node ./create-manifest.js')
         readJSON(file: 'dist/manifest.json')
     }
-    buildType = { (env.BRANCH_NAME == 'master' || env.BRANCH_NAME.startsWith('release/')) ? 'MAINLINE' : 'FEATURE' }
-    publishPackage = { 'prod' }
-    testJob = 'spigot-tests-webrtcsdk'
 
-    buildStep = {
-        sh('''
-            npm i -g npm@7
+    testJob = 'no-tests'
+    // testJob = 'spigot-tests-webrtcsdk'
```

Comment:
this should be uncommented unless there's a reason why this *cant* run.

Reply:
Good point. This calls the test job differently than the other so I will need to update the spigot jobs.